pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -992,40 +1010,40 @@ func (o *MustGatherOptions) waitForGatherContainerRunning(pod *corev1.Pod) error
 	})
 }
 
-func (o *MustGatherOptions) getNamespace() (*corev1.Namespace, func(), error) {
+func (o *MustGatherOptions) getNamespace(ctx context.Context) (*corev1.Namespace, func(context.Context), error) {
 	if o.RunNamespace == "" {
-		return o.createTempNamespace()
+		return o.createTempNamespace(ctx)
 	}
 
-	ns, err := o.Client.CoreV1().Namespaces().Get(context.TODO(), o.RunNamespace, metav1.GetOptions{})
+	ns, err := o.Client.CoreV1().Namespaces().Get(ctx, o.RunNamespace, metav1.GetOptions{})
 	if err != nil {
 		return nil, nil, fmt.Errorf("retrieving namespace %q: %w", o.RunNamespace, err)
 	}
 
-	return ns, func() {}, nil
+	return ns, func(context.Context) {}, nil
 }
 
-func (o *MustGatherOptions) createTempNamespace() (*corev1.Namespace, func(), error) {
-	ns, err := o.Client.CoreV1().Namespaces().Create(context.TODO(), newNamespace(), metav1.CreateOptions{})
+func (o *MustGatherOptions) createTempNamespace(ctx context.Context) (*corev1.Namespace, func(context.Context), error) {
+	ns, err := o.Client.CoreV1().Namespaces().Create(ctx, newNamespace(), metav1.CreateOptions{})
 	if err != nil {
 		return nil, nil, fmt.Errorf("creating temp namespace: %w", err)
 	}
 	o.PrinterCreated.PrintObj(ns, o.LogOut)
 
-	crb, err := o.Client.RbacV1().ClusterRoleBindings().Create(context.TODO(), newClusterRoleBinding(ns), metav1.CreateOptions{})
-	if err != nil {
-		return nil, nil, fmt.Errorf("creating temp clusterRoleBinding: %w", err)
-	}
-	o.PrinterCreated.PrintObj(crb, o.LogOut)
-
-	cleanup := func() {
-		if err := o.Client.CoreV1().Namespaces().Delete(context.TODO(), ns.Name, metav1.DeleteOptions{}); err != nil {
+	cleanup := func(ctx context.Context) {
```

Comment:
This would be really nice to have in a separate commit with the good explanation. But it is ok for this.

Reply:
What exactly needs explanation?

Reply:
I meant in a separate commit with a separate message about why do we change the order of clusterRole creation and namespace deletion.